Client/Utils/FaviconFetcher+Non-Bundled.swift

```diff
@@ -164,11 +159,16 @@ extension FaviconFetcher {
             } else {
                 return deferred.fill(Maybe(failure: FaviconError()))
             }
-            SDWebImageManager.shared.loadImage(with: iconURL, options: .continueInBackground, progress: nil) { (image, _, _, _, _, _) in
-                if let image = image {
-                    deferred.fill(Maybe(success: image))
-                } else {
-                    deferred.fill(Maybe(failure: FaviconError()))
+
+            if let iconURL = iconURL {
+                ImageLoadingHandler.shared.getImageFromCacheOrDownload(with: iconURL, limit: 0) { image, error in
+                    if error != nil || image == nil {
+                        deferred.fill(Maybe(failure: FaviconError()))
+                    }
+
+                    if let image = image {
+                        deferred.fill(Maybe(success: image))
+                    }
```

Comment:
nit: We could use guard here to image check twice. I added the suggestion but I'm not sure of the indentation
`guard let image = image, error == nil else {
      deferred.fill(Maybe(failure: FaviconError()))
      return
  }
  
 deferred.fill(Maybe(success: image))`

Reply:
oh good call, I missed this as part of my cleanup. let me guard the areas